src/queue/queue.service.ts

```diff
@@ -0,0 +1,114 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { Cron } from '@nestjs/schedule';
+import { Address } from '@planetarium/account';
+import { BencodexDictionary } from '@planetarium/bencodex';
+import { Currency, encodeFungibleAssetValue } from '@planetarium/tx/dist/assets';
+import { Job } from '@prisma/client';
+import { randomUUID } from 'crypto';
+import { PrismaService } from 'src/prisma/prisma.service';
+import { TxService } from 'src/tx/tx.service';
+
+const TX_ACTIONS_SIZE = 50;
+
+// FIXME: Get QUEUE_ADDRESS from external source (e.g., environment varibles).
+const QUEUE_ADDRESS = Address.fromHex("0x0000000000000000000000000000000000000000");
+
+// FIXME: Get NCG_MINTER from external source (e.g., environment varibles).
+const NCG_MINTER = Address.fromHex("0x47d082a115c63e7b58b1532d20e631538eafadde");
+
+@Injectable()
+export class QueueService {
+  private readonly logger = new Logger(QueueService.name);
+
+  constructor(private readonly prismaService: PrismaService, private readonly txService: TxService) {}
+
+  @Cron('00,10,20,30,40,50 * * * * *')
+  async handleCron() {
+    await this.prismaService.$transaction(async tx => {
+      const claimItemJobs: Job[] = await this.prismaService.job.findMany({
```

Comment:
```suggestion
      const claimItemJobs: Job[] = await tx.job.findMany({
```
I think it should be `tx`